packages/front-end/components/Settings/Team/PendingMemberList.tsx

```diff
@@ -0,0 +1,157 @@
+import { FC, useState } from "react";
+import { FaCheck, FaTimes, FaUserCheck } from "react-icons/fa";
+import { PendingMember } from "back-end/types/organization";
+import { roleHasAccessToEnv, useAuth } from "@/services/auth";
+import { datetime } from "@/services/dates";
+import ProjectBadges from "@/components/ProjectBadges";
+import DeleteButton from "@/components/DeleteButton/DeleteButton";
+import { useEnvironments } from "@/services/features";
+import MoreMenu from "@/components/Dropdown/MoreMenu";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import ChangeRoleModal from "@/components/Settings/Team/ChangeRoleModal";
+
+const PendingMemberList: FC<{
+  pendingMembers: PendingMember[];
+  mutate: () => void;
+  project: string;
+}> = ({ pendingMembers, mutate, project }) => {
+  const { apiCall } = useAuth();
+  const [roleModal, setRoleModal] = useState<string>("");
+  const { projects } = useDefinitions();
+  const environments = useEnvironments();
+
+  const roleModalUser = pendingMembers.find(
+    (member) => member.id === roleModal
+  );
```

Comment:
Minor detail but it could be more concise to make `roleModalUser` a state variable and setting it to a `PendingMember` to trigger the modal

```suggestion
  const [roleModalUser, setRoleModalUser] = useState<PendingMember | null>(null);
  const { projects } = useDefinitions();
  const environments = useEnvironments();
```